quidel_covidtest/delphi_quidel_covidtest/run.py

```diff
@@ -80,65 +95,68 @@ def run_module(params: Dict[str, Any]):
+    export_start_date = check_export_start_date(
+        export_start_date, export_end_date, export_day_range)
 
     first_date, last_date = df["timestamp"].min(), df["timestamp"].max()
 
-    # State Level
     data = df.copy()
     # Add prefix, if required
     sensors = add_prefix(SENSORS,
                          wip_signal=params["indicator"]["wip_signal"],
                          prefix="wip_")
-    smoothers = SMOOTHERS.copy()
+    smoothers = get_smooth_info(sensors, SMOOTHERS)
     for geo_res in NONPARENT_GEO_RESOLUTIONS:
         geo_data, res_key = geo_map(geo_res, data)
         geo_groups = geo_data.groupby(res_key)
-        for sensor in sensors:
-            logger.info("Generating signal and exporting to CSV",
-                        geo_res=geo_res,
-                        sensor=sensor)
-            if sensor.endswith(SMOOTHED_POSITIVE):
-                smoothers[sensor] = smoothers.pop(SMOOTHED_POSITIVE)
-            elif sensor.endswith(RAW_POSITIVE):
-                smoothers[sensor] = smoothers.pop(RAW_POSITIVE)
-            elif sensor.endswith(SMOOTHED_TEST_PER_DEVICE):
-                smoothers[sensor] = smoothers.pop(SMOOTHED_TEST_PER_DEVICE)
-            else:
-                smoothers[sensor] = smoothers.pop(RAW_TEST_PER_DEVICE)
-            state_df = generate_sensor_for_nonparent_geo(
-                geo_groups, res_key, smooth=smoothers[sensor][1],
-                device=smoothers[sensor][0], first_date=first_date,
-                last_date=last_date)
-            dates = create_export_csv(
-                state_df,
-                geo_res=geo_res,
-                sensor=sensor,
-                export_dir=export_dir,
-                start_date=export_start_date,
-                end_date=export_end_date)
-            if len(dates) > 0:
-                stats.append((max(dates), len(dates)))
+        for agegroup in AGE_GROUPS:
+            for sensor in sensors:
+                if agegroup == "total":
+                    sensor_name = sensor
+                else:
+                    sensor_name = "_".join([sensor, agegroup])
+                logger.info("Generating signal and exporting to CSV",
+                            geo_res=geo_res,
+                            sensor=sensor_name)
+                state_df = generate_sensor_for_nonparent_geo(
+                    geo_groups, res_key, smooth=smoothers[sensor][1],
+                    device=smoothers[sensor][0], first_date=first_date,
+                    last_date=last_date, suffix=agegroup)
+                dates = create_export_csv(
+                    state_df,
+                    geo_res=geo_res,
+                    sensor=sensor_name,
+                    export_dir=export_dir,
+                    start_date=export_start_date,
+                    end_date=export_end_date)
+                if len(dates) > 0:
+                    stats.append((max(dates), len(dates)))
 
```

Comment:
There's some hidden logic here that `geo_groups == geo_map("state", data).groupby("state_id")` . This should make it clearer

```suggestion
assert geo_res == "state"
```